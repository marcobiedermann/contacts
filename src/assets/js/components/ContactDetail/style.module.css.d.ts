declare const styles: {
  readonly details: string;
};
export = styles;
