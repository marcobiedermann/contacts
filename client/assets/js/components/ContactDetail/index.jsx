import PropTypes from 'prop-types';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import * as routes from '../../constants/routes';

const ContactDetail = props => {
  const { address, email, id, name, phone } = props;
  const { t } = useTranslation();

  return (
    <div>
      <header>
        <Link to={`${routes.CONTACTS}/${id}/edit`}>{t('common:edit')}</Link>
      </header>
      <h1>
        {name.firstName} {name.lastName}
      </h1>
      <address>
        <a
          href={`https://maps.google.com/?q=${encodeURIComponent(address.streetAddress)} ${address.zipCode} ${
            address.city
          } ${address.country}`}
          target="_blank"
          rel="noopener noreferrer"
        >
          {address.streetAddress}
          <br />
          {address.zipCode} {address.city}
          <br />
          {address.country}
        </a>
      </address>
      <ul>
        <li>
          <a href={`mailto:${email}`}>{email}</a>
        </li>
        <li>{phone}</li>
      </ul>
    </div>
  );
};

ContactDetail.propTypes = {
  address: PropTypes.shape(),
  email: PropTypes.string,
  id: PropTypes.string.isRequired,
  name: PropTypes.shape(),
  phone: PropTypes.string,
};

ContactDetail.defaultProps = {
  address: null,
  email: '',
  name: null,
  phone: '',
};

export default ContactDetail;
