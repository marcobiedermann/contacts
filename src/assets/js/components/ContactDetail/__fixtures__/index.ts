import { ContactDetailProps } from '..';
import address from '../../Address/__fixtures__';
import email from '../../Email/__fixtures__';

const props: ContactDetailProps = {
  id: '1',
  name: {
    firstName: 'John',
    lastName: 'Doe',
  },
  address,
  email: email.value,
  phone: '+1 234 56789',
};

export default props;
