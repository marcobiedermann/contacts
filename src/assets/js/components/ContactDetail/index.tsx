import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import styles from './style.module.css';

export interface ContactDetailProps {
  address?: {
    city: string;
    country: string;
    streetAddress: string;
    zipCode: string;
  };
  email?: string;
  id: string;
  name: {
    firstName: string;
    lastName: string;
  };
  phone?: string;
}

const ContactDetail: FC<ContactDetailProps> = (props) => {
  const { address, email, id, name, phone } = props;
  const { t } = useTranslation();

  return (
    <div>
      <header>
        <Link to={`/contacts/${id}/edit`}>{t('common:edit')}</Link>
      </header>
      {name && (
        <h1>
          {name.firstName} {name.lastName}
        </h1>
      )}
      <ul className={styles.details}>
        {phone && (
          <li>
            <a href={`tel:${phone}`}>{phone}</a>
          </li>
        )}
        {email && (
          <li>
            <a href={`mailto:${email}`}>{email}</a>
          </li>
        )}
        {address && (
          <li>
            <address>
              <a
                href={`https://maps.google.com/?q=${encodeURIComponent(address.streetAddress)} ${address.zipCode} ${
                  address.city
                } ${address.country}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                {address.streetAddress}
                <br />
                {address.zipCode} {address.city}
                <br />
                {address.country}
              </a>
            </address>
          </li>
        )}
      </ul>
    </div>
  );
};

export default ContactDetail;
