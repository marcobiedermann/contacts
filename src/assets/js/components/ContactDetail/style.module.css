.details {
  list-style: none;
  margin: 0;
  padding: 0;
}

.details li {
  padding-bottom: 0.5em;
  padding-top: 0.5em;
}

.details li + li {
  border-top: 1px solid #ededed;
}
