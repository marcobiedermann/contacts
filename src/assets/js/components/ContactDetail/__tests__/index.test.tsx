import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import ContactDetail from '..';
import props from '../__fixtures__';

describe('contact detail component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const { container } = render(
      <MemoryRouter>
        <ContactDetail {...props} />
      </MemoryRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
