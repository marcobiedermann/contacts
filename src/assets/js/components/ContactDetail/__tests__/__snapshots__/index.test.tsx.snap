// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact detail component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <header>
      <a
        href="/contacts/1/edit"
      >
        common:edit
      </a>
    </header>
    <h1>
      John Doe
    </h1>
    <ul
      class="details"
    >
      <li>
        <ul>
          <li>
            <div>
              <div>
                home
              </div>
              <div>
                <a
                  href="tel:+1 234 56789"
                >
                  +1 234 56789
                </a>
              </div>
            </div>
          </li>
        </ul>
      </li>
      <li>
        <ul>
          <li>
            <div>
              <div>
                home
              </div>
              <div>
                <a
                  href="mailto:john.doe@gmail.com"
                >
                  john.doe@gmail.com
                </a>
              </div>
            </div>
          </li>
        </ul>
      </li>
      <li>
        <ul>
          <li>
            <address>
              <div>
                home
              </div>
              <div>
                <a
                  href="https://maps.google.com/?q=Main%20Street 10001 New York City USA"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Main Street
                  <br />
                  10001 New York City
                  <br />
                  USA
                </a>
              </div>
            </address>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
